src/Adapter/BestSales/BestSalesProductSearchProvider.php

```diff
@@ -42,10 +42,16 @@ class BestSalesProductSearchProvider implements ProductSearchProviderInterface
      */
     private $translator;
 
+    /**
+     * @var SortOrderFactory
+     */
+    private $sortOrderFactory;
+
     public function __construct(
         TranslatorInterface $translator
     ) {
         $this->translator = $translator;
+        $this->sortOrderFactory = new SortOrderFactory($this->translator);
```

Comment:
This one is deprecated, use \PrestaShop\PrestaShop\Core\Product\Search\SortOrdersCollection

Reply:
@FabienPapet Fixed, thanks dude!